concepts/05 Widgets/DataGrid/25 Sorting/20 API/5 Custom Sorting.md

```diff
@@ -92,5 +143,92 @@ Implement a custom sorting routine using the [calculateSortValue](/api-reference
             [calculateSortValue]="customSortingFunction">
         </dxi-column>
     </dx-data-grid>
-    
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid :ref="dataGridRefKey">
+            <DxColumn
+                :calculate-sort-value="calculateSortValue"
+                data-field="Position"
+                sort-order="asc"
+            />
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxDataGrid, DxColumn } from 'devextreme-vue/data-grid';
+
+    const dataGridRefKey = "my-data-grid";
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn
+        },
+        data() {
+            return {
+                dataGridRefKey
+            };
+        },
+        methods: {
+            calculateSortValue(rowData) {
```

Comment:
You can move `calculateSortValue` to `data()`, and the DataGrid instance won't be needed. In this case, `this` will point to the column:

```js
data() {
    return {
        calculateSortValue: function(rowData) {
            if (rowData.Position === 'CEO')
                return this.sortOrder === 'asc' ? 'aaa' : 'zzz'; // CEOs are always displayed at the top  
            else
                return rowData.Position; // Others are sorted as usual
        }
    };
}

```